WordPress/src/main/java/org/wordpress/android/ui/suggestion/service/SuggestionService.java

```diff
@@ -97,11 +100,68 @@ public void run() {
         }.start();
     }
 
-    private void removeSiteIdFromRequestsAndStopServiceIfNecessary(Integer remoteBlogId) {
-        mCurrentlyRequestingSiteIds.remove(remoteBlogId);
+    private void removeSiteIdFromSuggestionRequestsAndStopServiceIfNecessary(Integer remoteBlogId) {
+        mCurrentlyRequestingSuggestionsSiteIds.remove(remoteBlogId);
 
         // if there are no requests being made, we want to stop the service
-        if (mCurrentlyRequestingSiteIds.isEmpty()) {
+        if (mCurrentlyRequestingSuggestionsSiteIds.isEmpty()) {
+            AppLog.d(AppLog.T.SUGGESTION, "stopping suggestion service");
+            stopSelf();
+        }
+    }
+
+    public void updateTags(final int remoteBlogId) {
+        if (mCurrentlyRequestingTagsSiteIds.contains(remoteBlogId)) {
+            return;
+        }
+        mCurrentlyRequestingTagsSiteIds.add(remoteBlogId);
+        RestRequest.Listener listener = new RestRequest.Listener() {
+            @Override
+            public void onResponse(JSONObject jsonObject) {
+                handleTagsUpdatedResponse(remoteBlogId, jsonObject);
+                removeSiteIdFromTagRequestsAndStopServiceIfNecessary(remoteBlogId);
+            }
+        };
+        RestRequest.ErrorListener errorListener = new RestRequest.ErrorListener() {
+            @Override
+            public void onErrorResponse(VolleyError volleyError) {
+                AppLog.e(AppLog.T.SUGGESTION, volleyError);
+                removeSiteIdFromTagRequestsAndStopServiceIfNecessary(remoteBlogId);
+            }
+        };
+
+        AppLog.d(AppLog.T.SUGGESTION, "suggestion service > updating tags for siteId: " + remoteBlogId);
+        String path = "/sites/" + remoteBlogId + "/tags";
+        WordPress.getRestClientUtils().get(path, listener, errorListener);
+    }
+
+    private void handleTagsUpdatedResponse(final int remoteBlogId, final JSONObject jsonObject) {
+        new Thread() {
+            @Override
+            public void run() {
+                if (jsonObject == null) {
+                    return;
+                }
+
+                JSONArray jsonTags = jsonObject.optJSONArray("tags");
+                List<Tag> tags = Tag.tagListFromJSON(jsonTags, remoteBlogId);
+                if (tags != null) {
+                    SuggestionTable.insertTagsForSite(remoteBlogId, tags);
+
+                    LocalBroadcastManager lbm = LocalBroadcastManager.getInstance(WordPress.getContext());
+                    Intent intent = new Intent(SuggestionService.ACTION_TAGS_LIST_UPDATED);
+                    intent.putExtra(SuggestionService.TAGS_LIST_UPDATED_EXTRA, remoteBlogId);
+                    lbm.sendBroadcast(intent);
```

Comment:
This event is broadcasted but never caught. We should catch it and update `mTagSuggestionAdapter` like it was done for name suggestion here for instance: https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/comments/CommentDetailFragment.java#L381-L385